src/main/java/com/ecommerce/ECommerceApp/model/Email.java

```diff
@@ -21,7 +23,7 @@ public void setEmailAddress(String emailAddress) {
         if (isValid(emailAddress)) {
             this.emailAddress = emailAddress;
         } else {
-            throw new IllegalArgumentException("Invalid email address: " + emailAddress);
+            throw new InvalidEmailException("Invalid email address: " + emailAddress);
```

Comment:
Implemented custom exception handling for invalid email to help make code more readable and clearer on what is being handled during an exception being thrown.

Source: https://www.baeldung.com/java-clean-code